blakeblackshear/frigate
# docs: move docs to docusaurus

# Live preview: https://frigate-nzwqyym4t.vercel.app/

## Motivation
Frigate has a fairly complex set of instructions for configuration with many options, requirements, and advanced features. It is also moving fairly quickly with some pretty important breaking changes, which makes it hard to find the correct documentation for the version that someone is using.

## Solution
Use a documentation site generated by [Docusaurus](https://v2.docusaurus.io). 

* Can be built and automatically deployed to `gh-pages`, hosted for free by GitHub. 
* Supports [versioned documentation](https://v2.docusaurus.io/docs/versioning)
* [Search](https://v2.docusaurus.io/docs/search)
* Automatically hashes builds so no stale cache issues

This patch just sets up the basics:
* Ran `npx @docusaurus/init@latest init docs classic`
* Moved a few images around
* Copy/pasted some of the `README.md` file into various pages to show how to set things up (see: `docs/docs/*/**.md` files)

<img width="1392" alt="Screen Shot 2021-01-20 at 4 38 06 PM" src="https://user-images.githubusercontent.com/33297/105258118-e5b66280-5b3d-11eb-903e-76186cb93596.png">

<img width="1392" alt="Screen Shot 2021-01-20 at 4 38 22 PM" src="https://user-images.githubusercontent.com/33297/105258133-ebac4380-5b3d-11eb-992a-9b55bb2baba2.png">


### Testing:
* Ensure you have node 12 or 14 installed
* `cd docs`
* `npm install`
* `npm run start` (your browser window should automatically open to `http://localhost:3000`)
* Edit any file in docs and the browser will auto-refresh

### In order to complete this work:
* [x] Move all current `README.md` contents to appropriate pages
* [x] Enable a GitHub workflow to automatically deploy docs ([example](https://github.com/paularmstrong/build-tracker/blob/main/.github/workflows/push.yml#L44-L60))
* [x] Update colors
* [x] Update main homepage
* [x] Remove (or update) contents of `docs/blog`
* [x] Update `README.md` to only include minimal relevant information, point to the GitHub hosted docs pages
* [ ] Once deployed, apply for [Algolia DocSearch](https://v2.docusaurus.io/docs/search)
